File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;
class Solution {
    boolean solution(String s) {
        Stack<Character> brackets = new Stack<>();
        for(int i = 0; i < s.length(); i++) {
            if(s.charAt(i) == '(') {
                brackets.push('(');
            } else if(s.charAt(i) == ')') {
                if(!brackets.isEmpty()) {
                    brackets.pop();
                }  else {
                    return false;
                }
            }
        }
     return !brackets.isEmpty() ? false : true;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
   let stack = [];
  
   for(const b of s) {
       if(b === "(") {
           stack.push(b);
       } else if(b === ")") {
           if(stack.length > 0 && stack[stack.length - 1] === "(") {
               stack.pop();
           } else {
               stack.push(b);
               break;
           }
       }
   }
   return stack.length === 0 ? true : false;
}